packages/cli/src/lib/plugins/clean.ts

```diff
@@ -0,0 +1,506 @@
+      return packageJson.dependencies?.metro || packageJson.devDependencies?.metro;
+    } catch {
+      return false;
+    }
+  }
+  
+  return false;
+}
+
+/**
+ * Cleans temporary directories that match a given pattern.
+ * @param pattern - The pattern to match temporary directory names
+ */
+function cleanTempDirectoryPattern(pattern: string): void {
+  const tmpDir = process.env['TMPDIR'] || process.env['TMP'] || '/tmp';
+  try {
+    const tmpDirContents = fs.readdirSync(tmpDir);
+    const matchingFiles = tmpDirContents
+      .filter(name => name.startsWith(pattern))
+      .map(name => path.join(tmpDir, name));
+    
+    for (const file of matchingFiles) {
+      if (fs.existsSync(file)) {
+        fs.rmSync(file, { recursive: true, force: true });
+      }
+    }
+  } catch (error) {
+    logger.debug(`${pattern} cache cleanup failed: ${error}`);
+  }
+}
+
+/**
+ * Creates an array of cleanup tasks for a specific project.
+ * @param projectRoot - The root directory of the project
+ * @param options - Clean options that affect task creation
+ * @returns Array of cleanup tasks with their configurations
+ */
+function createCleanupTasks(projectRoot: string, options: CleanOptions): CleanupTask[] {
+  const tasks: CleanupTask[] = [];
+
+  // Android cleanup
+  tasks.push({
+    name: 'android',
+    description: 'Android build caches (Gradle)',
+    enabled: true,
+    action: async () => {
+      const androidDir = path.join(projectRoot, 'android');
+      if (fs.existsSync(androidDir)) {
+        // Clean Gradle cache
+        const gradleDir = path.join(androidDir, '.gradle');
+        if (fs.existsSync(gradleDir)) {
+          fs.rmSync(gradleDir, { recursive: true, force: true });
+        }
+        
+        // Clean build directory
+        const buildDir = path.join(androidDir, 'build');
+        if (fs.existsSync(buildDir)) {
+          fs.rmSync(buildDir, { recursive: true, force: true });
+        }
+        
+        // Run gradle clean if gradlew exists
+        const gradlewPath = path.join(androidDir, 'gradlew');
+        if (fs.existsSync(gradlewPath)) {
+          try {
+            await spawn('sh', [gradlewPath, 'clean'], { cwd: androidDir });
+          } catch (error) {
+            logger.debug(`Gradle clean failed: ${error}`);
+          }
+        }
+      }
+    },
+  });
+
+  // CocoaPods cleanup
+  tasks.push({
+    name: 'cocoapods',
+    description: 'CocoaPods cache and Pods directory',
+    enabled: true,
+    action: async () => {
+      const iosDir = path.join(projectRoot, 'ios');
+      if (fs.existsSync(iosDir)) {
+        // Remove Pods directory
+        const podsDir = path.join(iosDir, 'Pods');
+        if (fs.existsSync(podsDir)) {
+          fs.rmSync(podsDir, { recursive: true, force: true });
+        }
+        
+        // Clean CocoaPods cache
+        try {
+          await spawn('pod', ['cache', 'clean', '--all'], { cwd: iosDir });
```

Comment:
we should first try with `bundle exec pod cache clean` and then fall back to `pod cache clean`